primitives/types/src/lib.rs

```diff
@@ -0,0 +1,120 @@
+use derive_more::{Add, Constructor, Deref, Display, From, Into, Mul};
+use parity_scale_codec::{Decode, Encode, MaxEncodedLen};
+use scale_info::TypeInfo;
+#[cfg(feature = "std")]
+use serde::{Deserialize, Serialize};
+
+mod data_lookup;
+pub use data_lookup::*;
+
+/// Raw Extrinsic with application id.
+#[derive(Clone, TypeInfo, Default, Encode, Decode)]
+#[cfg_attr(feature = "substrate", derive(sp_debug_derive::RuntimeDebug))]
+pub struct AppExtrinsic {
+	pub app_id: AppId,
+	pub data: Vec<u8>,
+}
+
+#[derive(
+	Clone,
+	Copy,
+	PartialEq,
+	Eq,
+	PartialOrd,
+	Ord,
+	Add,
+	From,
+	Deref,
+	TypeInfo,
+	Encode,
+	Decode,
+	Display,
+	Into,
+	Default,
+	MaxEncodedLen,
```

Comment:
`TypeInfo`, `Encode`, `Decode` & `MaxEncodedLen` could be under `substrate` feature too, right?

Reply:
The main use-case I'm thinking of is in a browser, where you still want to be able to decode these things. In those cases, you can't compile all of substrate, but you can compile `parity-scale-codec`.